src/shared/components/ChannelCover/ChannelCover.tsx

```diff
@@ -0,0 +1,154 @@
+import { AllChannelFieldsFragment } from '@/api/queries'
+import { BackgroundPattern } from '@/components'
+import { transitions } from '@/shared/theme'
+import { formatNumberShort } from '@/utils/number'
+import React, { useState } from 'react'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+import { HeaderTextField } from '..'
+import Icon from '../Icon'
+import Tooltip from '../Tooltip'
+import {
+  ChannelInfo,
+  CoverImage,
+  EditableOverlay,
+  EditCoverButton,
+  Header,
+  Media,
+  MediaWrapper,
+  RemoveCoverButton,
+  StyledAvatar,
+  StyledButton,
+  SubTitle,
+  SubTitlePlaceholder,
+  Title,
+  TitleContainer,
+  TitlePlaceholder,
+  TitleSection,
+} from './ChannelCover.style'
+
+type BasicChannelCoverProps = {
+  isFollowing?: boolean
+  channel?: AllChannelFieldsFragment | null
+  handleFollow?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+}
+type EditableProps =
+  | {
+      editable?: false
+      handleEditCover?: never
+      handleRemovecover?: never
+      handleChangeName?: never
+      changeNameHelperText?: never
+      changeNameError?: never
+      changeNameWarning?: never
+    }
+  | {
+      editable: true
+      handleEditCover?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+      handleRemovecover?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+      handleChangeName?: (e: React.ChangeEvent<HTMLInputElement>) => void
+      changeNameHelperText?: string
+      changeNameError?: boolean
+      changeNameWarning?: boolean
+    }
+
+export type ChannelCoverProps = BasicChannelCoverProps & EditableProps
+
+const ChannelCover: React.FC<ChannelCoverProps> = ({
+  channel,
+  handleFollow,
+  isFollowing,
+  editable,
+  handleRemovecover,
+  handleEditCover,
+  handleChangeName,
+  changeNameHelperText,
+  changeNameError,
+  changeNameWarning,
+}) => {
+  const showBgPattern = !channel?.coverPhotoUrl
+  const [overlayVisible, setoverlayVisible] = useState(false)
+
+  return (
+    <Header>
+      <MediaWrapper>
+        <Media>
+          <TransitionGroup>
+            <CSSTransition
+              key={showBgPattern ? 'pattern' : 'cover'}
+              timeout={parseInt(transitions.timings.loading)}
+              classNames={transitions.names.fade}
+            >
+              {showBgPattern ? (
+                <BackgroundPattern />
+              ) : (
+                <CoverImage editable={editable && overlayVisible} src={channel?.coverPhotoUrl!} />
```

Comment:
I believe it's better to solve it somehow without this assertion - something like adding union type with `null` in `CoverImageProps` and then do conditional styling in `CoverImage`. What do you think?